### PR TITLE
mold 1.2

### PR DESCRIPTION
**Pull request by pthariensflame**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by pthariensflame** [comment]
Should be ready!

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
Thanks @pthariensflame !!

File: Formula/mold.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Mold < Formula
   desc "Modern Linker"
   homepage "https://github.com/rui314/mold"
-  url "https://github.com/rui314/mold/archive/v1.1.1.tar.gz"
-  sha256 "47c5ddfe60beffc01da954191c819d78924e4d1eb96aeebfa24e1862cb3a33f9"
+  url "https://github.com/rui314/mold/archive/v1.2.tar.gz"
```
Comment (chenrui333) [inline Formula/mold.rb:4]: should this be `1.2.0` instead of `1.2`?

Comment (pthariensflame) [reply Formula/mold.rb:4]: You’d think so but at the time I used `brew bump-formula-pr` the `1.2.0` tag didn’t even exist.  🤷

Comment (pthariensflame) [reply Formula/mold.rb:4]: I’ll change it over now that it does.

Comment (pthariensflame) [reply Formula/mold.rb:4]:
```suggestion
  url "https://github.com/rui314/mold/archive/v1.2.0.tar.gz"
```

**Comment by pthariensflame** [comment]
What’s the failure about?

**Comment by chenrui333** [comment]
> What’s the failure about?

```
$ shasum -a 256 mold-1.2.*
f3fd3d0fbb143695ef33126f496eb9f0449aae1c26c764fee394d71ba1cd4310  mold-1.2.0.tar.gz
975ad37394e5c32661ee5da2817e5e8b8d40c6a73c78e23b30a57763c004a9f1  mold-1.2.tar.gz
```

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @pthariensflame!

File: Formula/mold.rb [hunk 2]
```diff
@@ -20,6 +20,7 @@ class Mold < Formula
 
   on_macos do
     depends_on "llvm" => [:build, :test] if DevelopmentTools.clang_build_version <= 1200
+    depends_on macos: :monterey
```
Comment (carlocab) [inline Formula/mold.rb:23]: Why are we restricting this to Monterey and above?

Comment (pthariensflame) [reply Formula/mold.rb:23]: Because it failed to build on Big Sur, or so I thought?  You can try and relax it if you like; maybe I misread the logs.

Comment (carlocab) [reply Formula/mold.rb:23]: It should've been fixed and not merged, then. CC @chenrui333

Comment (carlocab) [reply Formula/mold.rb:23]: Works fine in #99538.

In the future, let's be more careful about adding restrictions to the macOS version to existing formulae, particularly when this requirement is not documented upstream. Adding these restrictions leads to a bad user experience when doing `brew upgrade`.

**Pull request closed without merging**